### ISSUE TITLE
refactor(docs-infra): remove template type-checking work-around for `*ngFor`

### ISSUE THREAD

**Issue by gkalpak** [angular/angular#33051]
Before #31371, the `ngFor` typing issue could be worked around by using the non-null assertion operator (`!`) in templates:

```diff
-*ngFor="let section of filteredSections | async"
+*ngFor="let section of (filteredSections | async)!"
```

Once #31371 is released (i.e. in v9.0.0-next.10 or v9.0.0-rc.0), we should update angular.io and remove such work-arounds (see https://github.com/angular/angular/pull/32980#discussion_r331797547).

**Comment by Splaktar** [comment]
It's now available in `9.0.0-next.10`.

**Comment by Splaktar** [comment]
Glad that I saw this, I also removed this workaround in https://github.com/angular/material.angular.io/pull/648.
